src/coord/src/coord.rs

```diff
@@ -472,11 +472,25 @@ where
                 // about how it was built. If we start building multiple sinks and/or indexes
                 // using a single dataflow, we have to make sure the rebuild process re-runs
                 // the same multiple-build dataflow.
-                CatalogItem::Source(_) => {
+                CatalogItem::Source(source) => {
                     // Inform the timestamper about this source.
                     self.update_timestamper(entry.id(), true).await;
-                    let frontiers =
-                        self.new_frontiers(entry.id(), Some(0), self.logical_compaction_window_ms);
+                    let since_ts = {
+                        match &source.connector {
+                            SourceConnector::External { persist, .. } => {
+                                persist.as_ref().map(|p| p.primary_stream.since_ts)
+                            }
+                            _ => None,
```

Comment:
nit: seems like you could save yourself the unwrap and make this `Some(0)` for the default case (also below)

Reply:
Yep, I thought about that, but I wanted to keep it that way because the `Some(since_ts)` really is a weird way of turning the one `since_ts` into an iterator. I wanted to keep that signalling, though I can understand why it's weird.